api/src/main/scala/com/lightbend/lagom/internal/api/broker/TopicFactory.scala

```diff
@@ -15,3 +20,26 @@ import com.lightbend.lagom.javadsl.api.broker.Topic
 trait TopicFactory {
   def create[Message](topicCall: TopicCall[Message]): Topic[Message]
 }
+
+/**
+ * Provider for a topic factory.
+ *
+ * This layer of indirection is provided so that the ServiceClientImplementor doesn't have to directly depend on a
+ * TopicFactory, it can be optional.
```

Comment:
Would it be possible to get the same result with something like another default implementation of `TopicFactory` that throws an exception? That might simplify things.